### ISSUE TITLE
PHP8 Support

### ISSUE THREAD

**Issue by zlodes** [Ocramius/GeneratedHydrator#159]

**Comment by Ocramius** [comment]
If anyone has time to send patches to test compat, I'll gladly merge/release 👍

**Comment by zlodes** [comment]
Create PR to add PHP8 support for CodeGenerationUtils:
https://github.com/Ocramius/CodeGenerationUtils/pull/34

**Comment by Ocramius** [comment]
Upstream patch merged and `ocramius/code-generation-utils` is now compatible with 8.x as per Ocramius/CodeGenerationUtils#35.

This repo needs more work, since I see that some more tooling isn't compatible yet, most notably https://github.com/Roave/BetterReflection/issues/701

**Comment by pounard** [comment]
I'm forking this library, for the time being until it has an official release with php8 support show up.

The only remaining blocking dependency I had to remove for it to work is `roave/backward-compatibility-check` (including all other dev dependencies),I also had to upgrade `phpbench/phpbench` to 1.x. Aside of that, all unit tests are passing. No patches in this repository's code are necessary except the `composer.json` dependencies relaxing.

But phpbench is unable to run anymore, I think it has breaking changes in its configuration. For my use case, all I need is this library to work on the targeted env, so I'm not bothering with phpbench yet. If I manage to find some spare time in the next weeks, I'll look at it, but no promises.
